### DISCUSSION TITLE
Work with data while importing using Import-DbaCsv

### DISCUSSION THREAD

**Discussion by heliolino** [dataplat/dbatools#7031] [Ideas]
I tried to find anything related to documentation, examples, but nothing 
It is related to the ability to manipulate data while loading. 

working on data migration from Oracle to SQL, I had a situation where one column from a .csv is 
coming with leading zeros before the whole number. 
It will be good if during the **map columns** we could do some data manipulation and apply some rules to the data. 
Regex maybe? 

ex. one column with data like 00002342 from source
then after the change, it will be like 2342 

ps. in this scenario, is not possible to change the datatype on the destination. 

anything like: 

$columns = @{ [decimal]$csvfile.col1 = 'cid'			}
or 
$columns = @{ col1 -replace '^0+(?=[^.])' = 'cid'	}

This is one situation but opens a new world of opportunity

**Comment by wsmelton** [comment]
That has been discussed previously in #6676
